### PR TITLE
breaking: require one of message or type for creating a thread

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

Helps update to api v10, as type became required to provide.
Additionally adds some missing documentation around creating threads.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
Increased the scope of this pr slightly in order to remove an unused method in http.py

**Comment by shiftinv** [comment]
I somehow don't see how a few of these changes are required; the overloads and docs updates make sense, but the code changes don't seem required (and add unnecessary breaking changes) given the library already always provides the `type` field in requests to the API, which defaulted to `private_thread` before.

File: disnake/channel.py [hunk 1]
```diff
@@ -723,25 +752,43 @@ async def create_thread(
...
             Defaults to ``None``.
+
+            .. versionchanged:: 2.5
+
+                Cannot be provided with ``type``.
+
+        type: :class:`ChannelType`
+            The type of thread to create. If a ``message`` is passed then this parameter
+            is ignored, as a thread created with a message is always a public thread.
+            By default this creates a private thread if this is ``None``.
```
Comment (shiftinv) [inline disnake/channel.py:781]:
```suggestion
            The type of thread to create.
```

File: disnake/channel.py [hunk 2]
```diff
@@ -706,6 +707,34 @@ def get_thread(self, thread_id: int, /) -> Optional[Thread]:
...
         return self.guild.get_thread(thread_id)
 
+    @overload
+    async def create_thread(
+        self,
+        *,
+        name: str,
+        message: Snowflake = None,
+        auto_archive_duration: AnyThreadArchiveDuration = None,
+        invitable: bool = None,
```
Comment (shiftinv) [inline disnake/channel.py:717]:
```suggestion
```

**Review by shiftinv — APPROVED** [review]

**Pull request merged**